soroban-sdk/src/env.rs

```diff
@@ -483,15 +480,12 @@ impl Env {
         impl internal::storage::SnapshotSource for EmptySnapshotSource {
             fn get(
                 &self,
-                _key: &Rc<xdr::LedgerKey>,
-            ) -> Result<(Rc<xdr::LedgerEntry>, Option<u32>), soroban_env_host::HostError>
-            {
-                let err: internal::Error = (ScErrorType::Storage, ScErrorCode::MissingValue).into();
-                Err(err.into())
-            }
-
-            fn has(&self, _key: &Rc<xdr::LedgerKey>) -> Result<bool, soroban_env_host::HostError> {
```

Comment:
Removing the `has` seems like an unnecessary breakage of the env API. We could keep the `has` function whilst still gaining all the same benefits. Keeping the `has` can also be zero cost if we add a default impl of the has function that uses the result of the get fn.

Reply:
Env doesn't need the `has` function and it would be cleaner to not request it from the implementers. The initial interface proved to be problematic to implement as implementer would need to make non-obvious assumptions about behavior of `get` when entry is missing. Since I've changed the trait anyway, I don't see a good reason for keeping an extra method that is never used.